ot/gaussian.py

```diff
@@ -344,6 +344,177 @@ def empirical_bures_wasserstein_distance(xs, xt, reg=1e-6, ws=None,
         return W
 
 
+def bures_wasserstein_barycenter(m, C, weights=None, num_iter=1000, eps=1e-7, log=False):
+    r"""Return OT linear operator between samples.
+
+    The function estimates the optimal barycenter of the
+    empirical distributions. This is equivalent to resolving the fixed point
+     algorithm for multiple Gaussian distributions :math:`\left{\mathcal{N}(\mu,\Sigma)\right}_{i=1}^n`
+    :ref:`[1] <references-OT-mapping-linear-barycenter>`.
+
+    The barycenter still following a Gaussian distribution :math:`\mathcal{N}(\mu_b,\Sigma_b)`
+    where :
+
+    .. math::
+        \mu_b = \sum_{i=1}^n w_i \mu_i
+
+    And the barycentric covariance is the solution of the following fixed-point algorithm:
+
+    .. math::
+        \Sigma_b = \sum_{i=1}^n w_i \left(\Sigma_b^{1/2}\Sigma_i^{1/2}\Sigma_b^{1/2}\right)^{1/2}
+
+
+    Parameters
+    ----------
+    m : list of array-like (d,)
+        mean of each distribution
+    C : list of array-like (d,d)
+        covariance of each distribution
+    weights : array-like (n,), optional
+        weights for each distribution
+    num_iter : int, optional
+        number of iteration for the fixed point algorithm
+    eps : float, optional
+        tolerance for the fixed point algorithm
+    log : bool, optional
+        record log if True
+
+
+    Returns
+    -------
+    mb : (d,) array-like
+        mean of the barycenter
+    Cb : (d, d) array-like
+        covariance of the barycenter
+    log : dict
+        log dictionary return only if log==True in parameters
+
+
+    .. _references-OT-mapping-linear-barycenter:
+    References
+    ----------
+    .. [1] M. Agueh and G. Carlier, "Barycenters in the Wasserstein space",
+        SIAM Journal on Mathematical Analysis, vol. 43, no. 2, pp. 904-924,
+        2011.
+    """
+    nx = get_backend(*C, *m,)
+
+    mb = nx.mean(m)
+    Cb = nx.mean(C, axis=0)
+
+    if weights is None:
+        weights = nx.ones(len(C), type_as=C[0]) / len(C)
+    diff = 1000  # I don't know what is the best value for this
```

Comment:
I don't know what is the best practice to initialize the diff. I was using np.inf, but here I'm using the nx backend, which doesn't support inf.